src/services/reviewService.js

```diff
@@ -3,14 +3,21 @@ const productRepo = require('../repositories/productRepo');
 
 const create = async (productId, review) => {
     // find the product with the id
-    const product = await productRepo.findById(productId);
+    const product = await productRepo.findByIdWithReviews(productId);
     // save the incoming review
     const newReview = await reviewRepo.save(review);
+    product.rating = (product.rating * product.reviews.length + newReview.rating)
+    / (product.reviews.length + 1);
```

Comment:
This logic can also be abstracted inside our POJO, We can use the same POJO method  conditionally to update the rating `product.setAverageRating()`